dts/msm8916/msm8916-512mb-mtp.dts

```diff
@@ -0,0 +1,51 @@
+// SPDX-License-Identifier: GPL-2.0-only
+
+/dts-v1/;
+
+#include <skeleton.dtsi>
+#include <lk2nd.h>
+
+/ {
+	compatible = "qcom,msm8916-512mb-mtp", "qcom,msm8916";
+	qcom,board-id = <8 0x100>;
+	
+	zhihe-ufi-001c {
+		model = "ufi-001b/ufi-001c 4G Modem Stick";
+		compatible = "zhihe,ufi-001c", "qcom,msm8916", "lk2nd,device";
+		qcom,msm-id = <206 0>;
+
+		/* Register EDL button as HOME button */
+		/* This is the only button on this device */
+		lk2nd,keys = <KEY_HOME 37 (GPIO_PULL_DOWN | GPIO_ACTIVE_HIGH)>;
+	};
+	
+	zhihe-sp970 {
+		model = "sp970 4G Modem Stick";
+		compatible = "zhihe,sp970", "qcom,msm8916", "lk2nd,device";
+		qcom,msm-id = <248 0>;
+
+		/* Register EDL button as HOME button */
+		/* This is the only button on this device */
+		lk2nd,keys = <KEY_HOME 107 (GPIO_PULL_DOWN | GPIO_ACTIVE_HIGH)>;
+	};
+
+	zhihe-uz801 {
+		model = "uz801 4G Modem Stick";
+		compatible = "zhihe,uz801", "qcom,msm8916", "lk2nd,device";
+		qcom,msm-id = <249 0>;
```

Comment:
`249` stands for MSM8**1**16 instead of MSM8916, which doesn't belong to here (`msm8916-*.dts`).